php-checks/src/main/java/org/sonar/php/checks/SelfKeywordUsageCheck.java

```diff
@@ -51,10 +54,18 @@
   private static final String MESSAGE = "Use \"static\" keyword instead of \"self\".";
 
   private Deque<Boolean> isFinalClassStack = new ArrayDeque<>();
+
   private Deque<Set<String>> finalMethodsStack = new ArrayDeque<>();
 
+  private List<String> privateProperties;
+
+  private List<String> privateMethods;
+
   @Override
   public void visitClassDeclaration(ClassDeclarationTree tree) {
+    privateProperties = new LinkedList<>();
```

Comment:
@vilchik-elena You're right. As the reason for using stacks was not explained, I had figured out the issue of nested classes. 
I changed the implementation using stacks.